server.js

```diff
@@ -16,8 +24,97 @@ const welcomeMessage = {
 //Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
+
+let maxID = Math.max(...messages.map(c => c.id))
+
+app.get("/", function (request, response, next) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+// app.get("/messages", function (request, response) {
+//   response.send(welcomeMessage);
+// });
+
+
+app.post('/messages', (req, res,next) => {
+  if (!req.body.from || !req.body.text){   
+    res.status(400).send(`You must include  name and message in your request`)
+    return
+  } 
+  else{
+    // res.send(req.body)
+    const newMessage = {
+      "id": ++maxID,
+      "from": req.body.from,
+      "text": req.body.text,
+      "date": new Date()
+    }
+    messages.push(newMessage)
+      res.send(`<h1>The message has been added</h1> <button onclick="history.back()"> Go Back </button> `)
+    
+  }
+
+});
+
+//Show or read al messages
+app.get('/messages', (req,res)=>{
+  res.send(messages)
+} )
+
+
+//Read or show on message by id
+app.get('/message', (req,res)=>{
+  const found=messages.filter(message => message.id===parseInt(req.query.id))
+  if(found.length>0){
+    res.send(found)
+    }
+
+else{
```

Comment:
Using early termination you can clean this code up by removing the else statement

```
if(found.length>0){
    res.send(found)
    return
    }
res.status(400).json({msg: `no message with the id of ${req.query.id}`})

```